docs/source/docs/registry.md

```diff
@@ -0,0 +1,31 @@
+# Registry
+
+> This is a functionality under active development. A full documentation will be
+> released later.
+
+DeepReg is adapting the usage of the registry system recently, to facilitate the usage
+of custom functionalities including
+
+- neural network architecture
+- loss
+- optimizer
+- data pre-processing
+- data loader
+
+The registry is defined in `deepreg/registry.py`, where the class `Registry` maintains a
+dictionary mapping `(category, key)` to `value`. It also provides the
+`build_from_config` functionality, which allows creating a class instance from a config
+directly. This allows the simplification of the config so that each configuration file
+need to provide the name and necessary arguments.
```

Comment:
needs

Reply:
Can change line 18 to "This allows for the simplification of the config file such that each configuration only needs to provide the name of the class and the arguments to build the class for use. Documentation on Registry config will be released with issue #525 ."

Bit more comprehensive?

Reply:
i added, except the last bit about doc, as it's mentioned in the title.